client/src/components/App/Content/InvestigationForm/TabManagement/PersonalInfoTab/usePersonalInfoTab.ts

```diff
@@ -0,0 +1,92 @@
+                        houseNum: getUpdatedField(investigatedPatient.addressByAddress.houseNum, personalInfoStateContext.personalInfoData.address.houseNum)
+                    },
+                    relevantOccupation: getUpdatedField(investigatedPatient.occupation, personalInfoStateContext.personalInfoData.relevantOccupation),
+                    educationOccupationCity: investigatedPatient.occupation === 'משרד החינוך' ? getUpdatedField(investigatedPatient.subOccupationBySubOccupation.city, personalInfoStateContext.personalInfoData.educationOccupationCity): '',
+                    institutionName: getUpdatedField(investigatedPatient.subOccupation, personalInfoStateContext.personalInfoData.institutionName),
+                    otherOccupationExtraInfo: getUpdatedField(investigatedPatient.other_occupation_extra_info, personalInfoStateContext.personalInfoData.otherOccupationExtraInfo)
+                });
+                investigatedPatient.subOccupationBySubOccupation && setSubOccupationName(getUpdatedField(investigatedPatient.subOccupationBySubOccupation.displayName, ''));
+                setCityName(getUpdatedField(investigatedPatient.addressByAddress.cityByCity.displayName, ''));
+                setStreetName(getUpdatedField(investigatedPatient.addressByAddress.streetByStreet.displayName, ''));
+            }
+        })
+    }
+
+    const getSubOccupations = (parentOccupation: string) => {
+        axios.get('/personalDetails/getSubOccupations?parentOccupation=' + parentOccupation).then((res: any) => {
+            setSubOccupations(res && res.data && res.data.data && res.data.data.allSubOccupations.nodes.map((node: any) => {
+                return {
+                    id: node.id, 
+                    subOccupation: node.displayName
+                }
+            }));
+        });
+    }
+
+    const getEducationSubOccupations = (city: string) => {
+        axios.get('/personalDetails/getEducationSubOccupations?city=' + city).then((res: any) => {
+            setSubOccupations(res && res.data && res.data.data && res.data.data.allSubOccupations.nodes.map((node: any) => { 
+                return {
+                    id: node.id, 
+                    subOccupation: node.displayName, 
+                    street: node.street
+                }
+            }));
+        });
+    }
+
+    const getStreetsByCity = (cityId: string) => {
+        axios.get('/addressDetails/city/' + cityId + '/streets').then((res: any) => {
+            setStreets(res && res.data && res.data.map((node: any) => {
+                return {
+                    displayName: node.displayName,
+                    id: node.id
+                }
+            }));
+        });
+    }
+
+    const getUpdatedField = (recievedField: any, currentField: any): any => {
+        return (recievedField ? recievedField : currentField);
```

Comment:
Why would you have an option to keep pervious values after fetching exsisting data? This function is called when the component is mounted and no data is inserted

Reply:
all of my fields are initialized to be empty strings. i think if i didnt have this function and just spread the returned object from the server with undefined values it would cause problems. ill check but i doubt i can change this

Reply:
כותבים received

Reply:
לך תמות

Reply:
לדעתי היא צודקת